python/cuspatial/cuspatial/tests/test_point_linestring_distance.py

```diff
@@ -167,3 +167,121 @@ def test_mixed_geometry_series_raise(lhs, rhs):
 
     with pytest.raises(ValueError, match=".*must contain only.*"):
         pairwise_point_linestring_distance(lhs, rhs)
+
+
+def test_multipoint_multilinestring_sliced_pairs(
+    multipoint_generator, multilinestring_generator
+):
+    num_pairs = 2
+    max_num_linestring_per_multilinestring = 2
+    max_num_segments_per_linestring = 2
+    max_num_points_per_multipoint = 2
+
+    hpts = gpd.GeoSeries(
+        [*multipoint_generator(num_pairs, max_num_points_per_multipoint)]
+    )
+    hlines = gpd.GeoSeries(
+        [
+            *multilinestring_generator(
+                num_pairs,
+                max_num_linestring_per_multilinestring,
+                max_num_segments_per_linestring,
+            )
+        ]
+    )
+
+    gpts = from_geopandas(hpts)
+    glines = from_geopandas(hlines)
+
+    hslicepts = hpts[0:1]
+    hslicelines = hlines[0:1]
+    slicegpts = gpts[0:1]
+    sliceglines = glines[0:1]
+    got = pairwise_point_linestring_distance(slicegpts, sliceglines)
+    expected = hslicepts.distance(hslicelines)
+
+    cudf.testing.assert_series_equal(got, cudf.Series(expected))
+
+    hslicepts = hpts[1:2]
+    hslicelines = hlines[1:2]
+    slicegpts = gpts[1:2]
+    sliceglines = glines[1:2]
+    got = pairwise_point_linestring_distance(slicegpts, sliceglines)
+    expected = hslicepts.distance(hslicelines)
+
+    # Just dropping the index for now, that is not libcuSpatial's problem
+    got.index = [1]
+    cudf.testing.assert_series_equal(got, cudf.Series(expected))
+
+
+@pytest.mark.parametrize("slice_index", [0, 1, 2, 3, 4])
+def test_multipoint_multilinestring_sliced_many(
+    multipoint_generator, multilinestring_generator, slice_twenty, slice_index
+):
```

Comment:
If you make changes according to my suggestion of fixture above, you don't need `slice_index`.

Reply:
This reflects a pattern that I'm trying to build for some other test cases as well. There are a few tests where I use premade slicing patterns, then iterate over them to get a fair coverage of ways a column can be sliced. In those tests I just rewrite the same slice group a few different times inside of the test file, because you can't parameterize a feature in pytest. I realized while writing this that I could put the slices in `conftest.py` as a fixture, then simply iterate over the fixture using a `@pytest.mark.parametrize` over a range, as here. The goal is not fundamentally to iterate the segments of `0,4...4,8...` but to iterate over any predefined slicing range. Given our limited resources I didn't opt to refactor those tests yet, but I'd like to leave this, unless you know of a better way to iterate over a fixture in multiple tests.

Reply:
> The goal is not fundamentally to iterate the segments of 0,4...4,8... but to iterate over any predefined slicing range.

If you need other slicing range, can't you just create other fixtures?